File: Python/1502. Can Make Arithmetic Progression From Sequence.py
class Solution:
    def canMakeArithmeticProgression(self, arr: List[int]) -> bool:
        arr.sort()
        temp = arr[1] - arr[0]
        for i in range(len(arr) - 1):
            if arr[i + 1] - arr[i] != temp: return False
        return True

File: C/1502. Can Make Arithmetic Progression From Sequence.c
void sort(int* arr, int arrSize) {
    for (int i = 1; i < arrSize; i++) {
        int temp = arr[i], j = i;

        while (j > 0 && arr[j - 1] > temp) {
            arr[j] = arr[j - 1];
            j--;
        }

        arr[j] = temp;
    }
}

bool canMakeArithmeticProgression(int* arr, int arrSize){
    sort(arr, arrSize);

    int dif = arr[1] - arr[0];

    for (int i = 1; i < arrSize; i++) {
        if (arr[i] - arr[i - 1] != dif) return 0;
    }

    return 1;
}
